packages/react-server-cli/src/compileClient.js

```diff
@@ -0,0 +1,222 @@
+//const logger = logging.getLogger(__LOGGER__);
+
+// compiles the routes file for browser clients using webpack.
+// returns a tuple of { compiler, serverRoutes }. compiler is a webpack compiler
+// that is ready to have run called, and serverRoutes is a path to the transpiled
+// server routes file, which can be required and passed in to
+// reactServer.middleware().
+// TODO: add options for sourcemaps.
+export default (routes,{
+	workingDir = "./__clientTemp",
+	routesDir = ".",
+	outputDir = workingDir + "/build",
+	outputUrl = "/static/",
+	hot = true,
+	minify = false,
+} = {}) => {
+	const workingDirAbsolute = path.resolve(process.cwd(), workingDir);
+	mkdirp.sync(workingDirAbsolute);
+	const outputDirAbsolute = path.resolve(process.cwd(), outputDir);
+	mkdirp.sync(outputDirAbsolute);
+
+	const routesDirAbsolute = path.resolve(process.cwd(), routesDir);
+
+	// for each route, let's create an entrypoint file that includes the page file and the routes file
+	let bootstrapFile = writeClientBootstrapFile(workingDirAbsolute);
+	const entrypointBase = hot ? [`webpack-dev-server/client?${outputUrl}`,"webpack/hot/only-dev-server"] : [];
+	let entrypoints = {};
+	for (let routeName of Object.keys(routes.routes)) {
+		let route = routes.routes[routeName];
+		let formats = normalizeRoutesPage(route.page);
+		for (let format of Object.keys(formats)) {
+			const absolutePathToPage = path.resolve(routesDirAbsolute, formats[format]);
+
+			entrypoints[`${routeName}${format !== "default" ? "-" + format : ""}`] = [
+				...entrypointBase,
+				bootstrapFile,
+				absolutePathToPage,
+			];
+		}
+	}
+
+	// now rewrite the routes file out in a webpack-compatible way.
+	const serverRoutes = writeWebpackCompatibleRoutesFile(routes, routesDir, workingDirAbsolute, outputUrl, false);
+	writeWebpackCompatibleRoutesFile(routes, routesDir, workingDirAbsolute, outputUrl, true);
+
+	// finally, let's pack this up with webpack.
+	return {
+		serverRoutes,
+		compiler: webpack(packageCodeForBrowser(entrypoints, outputDirAbsolute, outputUrl, hot, minify)),
+	};
+}
+
+const packageCodeForBrowser = (entrypoints, outputDir, outputUrl, hot, minify) => {
+	const extractTextLoader = require.resolve("./NonCachingExtractTextLoader") + "?{remove:true}!css-loader";
+	let webpackConfig = {
+		entry: entrypoints,
+		output: {
+			path: outputDir,
+			publicPath: outputUrl,
+			filename: "[name].bundle.js",
+			chunkFilename: "[id].bundle.js",
+		},
+		module: {
+			loaders: [
+				{
+					test: /\.jsx?$/,
+					loader: "babel",
+					exclude: /node_modules/,
+				},
+				{
+					test: /.css$/,
+					loader: extractTextLoader,
+					exclude: /node_modules/,
+				},
+			],
+		},
+		plugins: [
+			new ExtractTextPlugin("[name].css"),
+		],
+	};
+
+	if (minify) {
+		webpackConfig.plugins = [
+			...webpackConfig.plugins,
+			new webpack.DefinePlugin({
+				'process.env': {NODE_ENV: '"production"'},
+			}),
+			// TODO: should this be done as babel plugin?
```

Comment:
Probably not?  you don't want to uglify server side, and I'd like to try to keep the babel config as similar as possible on the server and client (I _think_ we should be transpiling them both pretty aggressively, since V8 features often start out pretty slow, and we want to support the oldest browsers reasonably possible)